storagenode/storagenodedb/infodb.go

```diff
@@ -38,11 +36,9 @@ func newInfo(path string) (*InfoDB, error) {
 	if err != nil {
 		return nil, ErrInfo.Wrap(err)
 	}
-
 	dbutil.Configure(db, mon)
 
 	infoDb := &InfoDB{db: db}
-	infoDb.pieceinfo = pieceinfo{InfoDB: infoDb}
 	infoDb.bandwidthdb = bandwidthdb{InfoDB: infoDb}
```

Comment:
this also

Reply:
The bandwidthdb is still doing the caching stuff, and so the infodb still needs to be returning the same one every time. I just removed the text zero initializing a bunch of fields to their zero values.